File: scripts/perl/greeting.pl
use strict;
use warnings;

use Time::HiRes 'time';
use Config::Tiny;

$| = 1;
my $config = Config::Tiny->read( '../etc/main.conf' );
my $DEBUG = $config->{global}->{DEBUG};

printf "Perl (greeting)%s...%s\n",
       $DEBUG ? ' in DEBUG mode' : '',
       $DEBUG ? '' : ' please wait';

my ($i, $l_greeting) = (0, 0);
my $greeting = $config->{greeting}->{GREETING};
my $final = $config->{greeting}->{FINAL};
my $t0 = time;

while ($i < $final) {
    printf "\r%s...", $greeting;
    $DEBUG and printf " %.1f%%", $i*100/$final;
    $l_greeting += length($greeting);
    $i++;
}
print "\n";

printf "'%s' %d times, total length %d\n",
    $greeting, $final, $l_greeting;

printf "time total: %.3f seconds\n", time - $t0;

File: scripts/python/greeting.py
import configparser
import time

config = configparser.ConfigParser()
config.read('../etc/main.conf')
debug = int(config['global']['DEBUG'])

if __name__ == "__main__":
    if debug:
        print("Python (greeting) in DEBUG mode...")
    else:
        print("Python (greeting)... please wait")

    t0, l_greeting, i = time.time(), 0, 0
    greeting = config['greeting']['GREETING']
    final = int(config['greeting']['FINAL'])

    while i < final:
        print("\r%s..." %( greeting ), end='', flush=True )
        debug and print(" %.1f%%" %( i*100 / final ), end='', flush=True )
        l_greeting += len(greeting)
        i += 1
    print()

    print("'%s' %d times, total length %d" %( greeting, final, l_greeting ) )
    print("time total: %.3f seconds" %( time.time() - t0 ) )
